lib/matplotlib/pyplot.py

```diff
@@ -1299,13 +1299,13 @@ def subplot2grid(shape, loc, rowspan=1, colspan=1, fig=None, **kwargs):
     -----
     The following call ::
 
-        subplot2grid(shape, loc, rowspan=1, colspan=1)
+        subplot2grid((nrows, ncols), (row, col), rowspan, colspan)
 
     is identical to ::
 
-        gridspec = GridSpec(shape[0], shape[1])
-        subplotspec = gridspec.new_subplotspec(loc, rowspan, colspan)
-        subplot(subplotspec)
+        fig = gcf()
+        fig.add_subplot(
+            fig.add_gridspec(nrows, ncols)[row:row+rowspan, col:col+colspan])
```

Comment:
Chaining these seems a bit confusing.
```suggestion
        gs = fig.add_gridspec(nrows, ncols)
        fig.add_subplot(gs[row:row+rowspan, col:col+colspan])
```

Reply:
sure